rust/src/detect/minversion.rs

```diff
@@ -0,0 +1,156 @@
+/* Copyright (C) 2023 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use nom7::bytes::complete::is_a;
+use nom7::character::complete::{char, digit1};
+use nom7::combinator::{map_opt, opt};
+use nom7::IResult;
+
+use std::ffi::CStr;
+
+fn min_version_check<'a>(rv: &'a str, sv: &'a str) -> IResult<&'a str, bool> {
```

Comment:
This mixes version parsing and version comparison.  Could we separate those? Perhaps parsing to an object, `SuricataVersion`, then comparing them?  The main function that others would call could stay the same though.